fluent-dom/src/localization.js

```diff
@@ -17,21 +17,25 @@ export default class Localization {
    *
    * @returns {Localization}
    */
-  constructor(resourceIds, generateMessages) {
+  constructor(resourceIds = [], generateMessages) {
     this.resourceIds = resourceIds;
     this.generateMessages = generateMessages;
-    this.ctxs =
-      new CachedAsyncIterable(this.generateMessages(this.resourceIds));
+    if (resourceIds.length) {
```

Comment:
Do we need this check here and in `onChange`? `generateMessages` and `CachedAsyncIterable` should work with all iterables, including empty ones. I think I'd prefer to not perform the check to signal that empty `resourceIds` is a valid scenario and not a special case.